test/core/integration/concrete/lockup/burn/burn.t.sol

```diff
@@ -2,177 +2,142 @@
+    function test_RevertWhen_CallerSender()
         external
         whenNoDelegateCall
         givenNotNull
-        givenDepletedStream(streamId)
-        whenAuthorizedCaller
+        givenDepletedStream(lockup, defaultStreamId)
+        whenCallerNotRecipient
     {
-        // Burn the NFT so that it no longer exists.
-        lockup.burn(streamId);
-
-        // Run the test.
-        vm.expectRevert(abi.encodeWithSelector(IERC721Errors.ERC721NonexistentToken.selector, streamId));
-        lockup.burn(streamId);
-    }
-
-    modifier givenNFTExists() {
-        _;
+        resetPrank({ msgSender: users.sender });
+        vm.expectRevert(
+            abi.encodeWithSelector(Errors.SablierLockup_Unauthorized.selector, defaultStreamId, users.sender)
+        );
+        lockup.burn(defaultStreamId);
     }
 
-    function test_GivenNonTransferableNFT()
+    function test_WhenCallerApprovedThirdParty()
         external
         whenNoDelegateCall
         givenNotNull
-        givenDepletedStream(notTransferableStreamId)
-        whenAuthorizedCaller
-        givenNFTExists
+        givenDepletedStream(lockup, defaultStreamId)
+        whenCallerNotRecipient
     {
-        // It should emit a {MetadataUpdate} event.
-        vm.expectEmit({ emitter: address(lockup) });
-        emit MetadataUpdate({ _tokenId: notTransferableStreamId });
+        resetPrank({ msgSender: users.recipient });
 
-        // Burn the NFT.
-        lockup.burn(notTransferableStreamId);
+        // Approve a third party to burn the NFT.
+        lockup.approve(users.alice, defaultStreamId);
+
+        // Make the third party the caller in this test.
+        resetPrank({ msgSender: users.alice });
 
         // It should burn the NFT.
-        vm.expectRevert(abi.encodeWithSelector(IERC721Errors.ERC721NonexistentToken.selector, notTransferableStreamId));
-        lockup.getRecipient(notTransferableStreamId);
+        _test_Burn(defaultStreamId);
     }
 
-    modifier givenTransferableNFT() {
-        _;
+    function test_RevertGiven_NFTNotExist()
+        external
+        whenNoDelegateCall
+        givenNotNull
+        givenDepletedStream(lockup, defaultStreamId)
+        whenCallerRecipient(users.recipient)
+    {
+        // Burn the NFT so that it no longer exists.
+        lockup.burn(defaultStreamId);
+
+        // Run the test.
+        vm.expectRevert(abi.encodeWithSelector(IERC721Errors.ERC721NonexistentToken.selector, defaultStreamId));
+        lockup.burn(defaultStreamId);
     }
 
-    function test_WhenCallerApprovedThirdParty()
+    /// @dev For this test, we will skip the `givenDepletedStream` modifier given the `notTransferableStreamId` is not
+    /// yet created.
+    function test_GivenNonTransferableNFT()
         external
         whenNoDelegateCall
         givenNotNull
-        givenDepletedStream(streamId)
-        whenAuthorizedCaller
+        /*  givenDepletedStream(lockup, notTransferableStreamId) */
```

Comment:
- Why its OK to skip `givenDepletedStream` branch for this test? 
- If its OK to skip `givenDepletedStream` for this test, why mention these comments? Are we going to test `givenDepletedStream` status anywhere else?

Reply:
have you seen the `dev` natspec?

```solidity
/// @dev For this test, we will skip the `givenDepletedStream` modifier given the `notTransferableStreamId` is not
/// yet created.
```

the problem is that this modifier has extra logic:

```solidity
modifier givenDepletedStream(ISablierLockup lockup, uint256 streamId) {
    vm.warp({ newTimestamp: lockup.getEndTime(streamId) });
    lockup.withdrawMax({ streamId: streamId, to: lockup.getRecipient(streamId) });
    _;
}
```


and passing `notTransferableStreamId` which is not created before this function would cause to revert.

one idea is to declare this id in `Integration_Test` contract (similar to defaultStreamId) and initialize this stream in `Lockup<Model>_Integration_Shared_Test` contracts but i am not sure if we need this stream anywhere else

> Are we going to test givenDepletedStream status anywhere else?

not sure what you mean, in order to burn the NFT we need to have a a depleted status, each non-reverting test must be depleted (this modifier)

Reply:
> have you seen the dev natspec?

Yes I have.

> one idea is to declare this id in Integration_Test contract (similar to defaultStreamId) and initialize this stream

How about creating `notTransferableStreamId` in `setUp` of this contract? Can that fix it?

> not sure what you mean, in order to burn the NFT we need to have a a depleted status, each non-reverting test must be depleted

Yes so we should think of including `givenDepletedStream` because we need that in order to call `burn`.

Reply:
> How about creating notTransferableStreamId in setUp of this contract? Can that fix it?

we can't create streams in `setUp` as the LD/LL/LT are not yet deployed at its level

```mermaid
flowchart TD
    FunctionName_LockupModel_Integration_TypeTest --> LockupModel_Integration_Shared_Test
    FunctionName_LockupModel_Integration_TypeTest --> FunctionName_Integration_TypeTest
    LockupModel_Integration_Shared_Test --> Integration_Test
    FunctionName_Integration_TypeTest --> Integration_Test
    Integration_Test --> Base_Test
    Base_Test --> Utils

    %% Adding a subgraph as a note
    subgraph note [ ]
        direction TB
        note_text(["Here the contracts still don't exist"])
    end

    note -.-> FunctionName_Integration_TypeTest
```

the tests are run in the child contracts, so for them it is not problem

Reply:
I didn't understand. Are you saying we cannot initialize `notTransferableStreamId` in `Burn_Integration_Concrete_Test.setUp()`? Ain't LD/LL/LT deployed in Base test?

Reply:
https://github.com/sablier-labs/v2-core/pull/1058/commits/7cc0f9dbbeb5f2608c7440b572769339759e0433